docs/pages/modules/module-api.md

```diff
@@ -0,0 +1,381 @@
+---
+title: Module API
+---
+
+import { CodeBlocksTable } from '~/components/plugins/CodeBlocksTable';
+
+Expo Module API is an abstraction layer of top of React Native modules that helps you build native modules in an easy to use and convenience way. As opposed to standard React Native modules, Expo Modules ecosystem is designed from the ground up to be used with only modern native languages — Swift and Kotlin. After many years of maintaining a bunch of various native modules as part of Expo SDK, we have found out that many issues were caused by not handling null values or using the wrong types. The lack of optional types and Objective-C being too dynamic, makes most bugs unable to get caught in the compile time — which would be caught in modern languages.
+
+Another big pain point that we have encountered is the validation of arguments passed from JavaScript to native functions. This is especially painful when it comes to `NSDictionary` or `ReadableMap`, where the type of values is unknown in runtime and each property needs to be validated separately by yourself. A valuable feature of the Expo Modules API is that it has full knowledge of the argument types the native function expects, so it can pre-validate and convert the arguments for you! The dictionaries can be represented as native structs that we call [Records](#records). Knowing the argument types, it is also possible to [automatically convert arguments](#convertibles) to some platform-specific types (e.g. `{ x: number, y: number }` or `[number, number]` can be translated to CoreGraphics's `CGPoint` for your convenience).
+
+## Get Started
+
+1️⃣ Firstly make sure that you [set up your library as an Expo module](overview). When it's set, create Swift and Kotlin files from the below templates.
+
+<CodeBlocksTable>
+
+```swift
+import ExpoModulesCore
+
+public class MyModule: Module {
+  public func definition() -> ModuleDefinition {
+    // Definition components go here
+  }
+}
+```
+
+```kotlin
+package my.module.package
+
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+
+class MyModule : Module() {
+  override fun definition() = ModuleDefinition {
+    // Definition components go here
+  }
+}
+```
+
+```
+// `import` from Kotlin seems to be breaking syntax highlighting in VSCode :O
+// Adding another code block solves the issue.
+```
+
+</CodeBlocksTable>
+
+2️⃣ Make sure that your module class is included in `modulesClassNames` in the [module config](module-config).
+
+<CodeBlocksTable tabs={["expo-module.config.json"]}>
+
+```json
+{
+  "ios": {
+    "modulesClassNames": ["MyModule"]
+  },
+  "android": {
+    "modulesClassNames": ["my.module.package.MyModule"]
+  }
+}
+```
+
+</CodeBlocksTable>
+
+3️⃣ On iOS you also need to run `pod install` to properly link the new class. On Android it will be linked automatically before building. Now you are ready to go and add some definition components in there!
```

Comment:
```suggestion

### 3. Ensure the module is linked

On iOS you also need to run `pod install` to properly link the new class. On Android it will be linked automatically before building. Now you are ready to go and add some definition components in there!
```